### ISSUE TITLE
BIFReader Not Working on Wiki Example

### ISSUE THREAD

**Issue by vsocrates** [pgmpy/pgmpy#1804]
### Subject of the issue
The BIFReader class isn't working on the `dog` example on the Wiki.

### Your environment
* pgmpy version: 0.1.25
* Python version: 3.10.14
* Operating System: Red Hat Enterprise Linux release 8.8

### Steps to reproduce
I attempted to run this example and it doesn't work: https://pgmpy.org/readwrite/bif.html#pgmpy.readwrite.BIF.BIFReader

[dog-problem.bif.zip](https://github.com/user-attachments/files/16460997/dog-problem.bif.zip)

The specific code I ran was:

```py
reader = BIFReader("./dog-problem.bif")
```

### Expected behaviour
Create the BIFReader class and allow interaction with a pgmpy BN.

### Actual behaviour
I got the following error:

```py
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Cell In[15], line 5
      2 from pgmpy.readwrite import BIFReader
      4 # reader = BIFReader("./sampleR_code/fitted_hc_v2.bif")
----> 5 reader = BIFReader("./sampleR_code/dog-problem.bif")

File [~/.conda/envs/bnlearn/lib/python3.10/site-packages/pgmpy/readwrite/BIF.py:99](https://ood-mccleary.ycrc.yale.edu/node/r813u15n08.mccleary.ycrc.yale.edu/25279/lab/tree/Documents/Uncertainty_Network/~/.conda/envs/bnlearn/lib/python3.10/site-packages/pgmpy/readwrite/BIF.py#line=98), in BIFReader.__init__(self, path, string, include_properties, n_jobs)
     97     self.variable_properties = self.get_property()
     98 self.variable_parents = self.get_parents()
---> 99 self.variable_cpds = self.get_values()
    100 self.variable_edges = self.get_edges()

File [~/.conda/envs/bnlearn/lib/python3.10/site-packages/pgmpy/readwrite/BIF.py:325](https://ood-mccleary.ycrc.yale.edu/node/r813u15n08.mccleary.ycrc.yale.edu/25279/lab/tree/Documents/Uncertainty_Network/~/.conda/envs/bnlearn/lib/python3.10/site-packages/pgmpy/readwrite/BIF.py#line=324), in BIFReader.get_values(self)
    305 def get_values(self):
    306     """
    307     Returns the CPD of the variables present in the network
    308 
   (...)
    323                         [0.4, 0.95]])}
    324     """
--> 325     cpd_values = Parallel(n_jobs=self.n_jobs)(
    326         delayed(self._get_values_from_block)(block)
    327         for block in self.probability_block()
    328     )
    330     variable_cpds = {}
    331     for var_name, arr in cpd_values:

File [~/.conda/envs/bnlearn/lib/python3.10/site-packages/joblib/parallel.py:1918](https://ood-mccleary.ycrc.yale.edu/node/r813u15n08.mccleary.ycrc.yale.edu/25279/lab/tree/Documents/Uncertainty_Network/~/.conda/envs/bnlearn/lib/python3.10/site-packages/joblib/parallel.py#line=1917), in Parallel.__call__(self, iterable)
   1916     output = self._get_sequential_output(iterable)
   1917     next(output)
-> 1918     return output if self.return_generator else list(output)
   1920 # Let's create an ID that uniquely identifies the current call. If the
   1921 # call is interrupted early and that the same instance is immediately
   1922 # re-used, this id will be used to prevent workers that were
   1923 # concurrently finalizing a task from the previous call to run the
   1924 # callback.
   1925 with self._lock:

File [~/.conda/envs/bnlearn/lib/python3.10/site-packages/joblib/parallel.py:1847](https://ood-mccleary.ycrc.yale.edu/node/r813u15n08.mccleary.ycrc.yale.edu/25279/lab/tree/Documents/Uncertainty_Network/~/.conda/envs/bnlearn/lib/python3.10/site-packages/joblib/parallel.py#line=1846), in Parallel._get_sequential_output(self, iterable)
   1845 self.n_dispatched_batches += 1
   1846 self.n_dispatched_tasks += 1
-> 1847 res = func(*args, **kwargs)
   1848 self.n_completed_tasks += 1
   1849 self.print_progress()

File [~/.conda/envs/bnlearn/lib/python3.10/site-packages/pgmpy/readwrite/BIF.py:302](https://ood-mccleary.ycrc.yale.edu/node/r813u15n08.mccleary.ycrc.yale.edu/25279/lab/tree/Documents/Uncertainty_Network/~/.conda/envs/bnlearn/lib/python3.10/site-packages/pgmpy/readwrite/BIF.py#line=301), in BIFReader._get_values_from_block(self, block)
    298         values_dict[tuple(states)] = vals
    299     for index, combination in enumerate(
    300         product(*[self.variable_states[var] for var in parents])
    301     ):
--> 302         arr[:, index] = values_dict[combination]
    303 return var_name, arr

KeyError: ('true',)
```

I'm actually having a different error with another BIF file created with `bnlearn` in R as well (I can create another Issue if need be). Also attached here: 
[fitted_hc_v2.bif.zip](https://github.com/user-attachments/files/16461016/fitted_hc_v2.bif.zip)

**Comment by ankurankan** [comment]
@vsocrates Thanks for creating the issue. I think the support for the BIF format used in the `dog` file was dropped quite a while back. I will update the documentation for that. We now use the file format used in bnlearn repository (https://www.bnlearn.com/bnrepository/) as the standard.

For the second model that you have shared, there were a couple of issues:
1. pgmpy didn't allow `.` in variable names but I have fixed it now.
2. Some of the state names in the file are missing (for example line 34, 40, etc.). I am not sure how that happened as the file was created using bnlearn. Is is possible for you to share the bnlearn object so that I can investigate further? Or if you know any potential reasons why this could have happened?
